velox/docs/develop/aggregate-functions.rst

```diff
@@ -152,13 +152,33 @@ A simple aggregation function is implemented as a class as the following.
     using IntermediateType = Array<Generic<T1>>;
     using OutputType = Array<Generic<T1>>;
 
+    // Define a struct for function-level states. Even if the aggregation function
```

Comment:
Can we make 'FunctionState' struct optional?

Reply:
We need to pass `FunctionState` as an input parameter for methods like `addInput`, `combine`, etc. If it's not defined, It will result in a compilation failure.